src/canisters/initializer/src/lib.rs

```diff
@@ -46,59 +46,137 @@ fn registry() -> Principal {
 }
 
 #[derive(CandidType, serde::Deserialize, Clone, Copy)]
-struct InitialeDeposits {
-    vault: u128,
-    db: u128,
-    proxy: u128,
+struct CycleManagement {
+    initial_supply: u128,
+    refueling_amount: u128,
+    refueling_threshold: u128,
+}
+
+#[derive(CandidType, serde::Deserialize, Clone, Copy)]
+struct CycleManagements {
+    refueling_interval: u64,
+    vault_intial_supply: u128,
+    indexer: CycleManagement,
```

Comment:
Why is the `CycleManagement` parameter an input, not a constant?
It looks dangerous since the deployer can configure this parameter a large value

Reply:
And we don't have to change its value for each canisters, isn't it?

Reply:
> Why is the CycleManagement parameter an input, not a constant?
> And we don't have to change its value for each canisters, isn't it?

I have no idea on that point now, so I made it changeable to reduce the cost of considering it.
I'm not sure the amount of cycle I can assume now will be enough, and the initial cost will be too much if make it more leeway.

Reply:
> It looks dangerous since the deployer can configure this parameter a large value

I agree there seems to be a risk, but I think it is not limited to here, and I would like to discuss it comprehensively at another time.